app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestViewModel.kt

```diff
@@ -45,15 +52,21 @@ class MeetingRequestViewModel(
     viewModelScope.launch {}
   }
 
+  fun setInitialValues(senderToken: String, senderUID: String, senderName: String) {
+    this.senderToken = senderToken
+    this.senderUID = senderUID
+    this.senderName = senderName
+  }
+
   /**
    * Sets the good token for our message, locally and in the profile's backend
    *
    * @param newToken the current token of the user to which we want to send the message
    */
   fun onRemoteTokenChange(newToken: String) {
     meetingRequestState = meetingRequestState.copy(targetToken = newToken)
-    profilesViewModel.getProfileByUid(MeetingRequestManager.ourUid ?: "null")
-    val currentProfile = profilesViewModel.selectedProfile.value
+    senderToken = newToken
```

Comment:
Can't you just do : 
```
fun onRemoteTokenChange(senderToken: String) {
    meetingRequestState = meetingRequestState.copy(targetToken = senderToken)
    ...
```

Reply:
Yes this changes the local token, but the token stored in our Profile firebase also needs to be updated. The onRemoteTokenChange function is only called in the MeetingRequestService in the onTokenChange, which is a function that Firebase can trigger at any time. The only purpose of this function is to keep the token in line with Firebase. It shouldn't be used otherwise

Reply:
You can have more details on this if you read the documentation of FirebaseMessagingService()

Reply:
I thought that you were instantiating senderToken here, thank you for the clarifications